### PR TITLE
Add upload cli

### PR DESCRIPTION
**Pull request by kylebarron**
Adds a cli command to upload a prebuilt mosaic to a backend

### CODE REVIEW

File: cogeo_mosaic/scripts/cli.py [hunk 1]
```diff
@@ -92,6 +92,19 @@ def create(
         click.echo(json.dumps(mosaicjson.dict(exclude_none=True)))
 
 
+@cogeo_cli.command(short_help="Upload mosaic definition to backend")
+@click.argument("input", type=click.File(mode="r"), default="-")
+@click.option("--url", type=str, help="URL to which the mosaic should be uploaded.")
```
Comment (vincentsarago) [inline cogeo_mosaic/scripts/cli.py:97]: this options should be `required`

File: cogeo_mosaic/scripts/cli.py [hunk 2]
```diff
@@ -92,6 +92,19 @@ def create(
...
+def upload(
+    file, url,
+):
```
Comment (vincentsarago) [inline cogeo_mosaic/scripts/cli.py:100]: does isort or black format this line like this ?

Comment (kylebarron) [reply cogeo_mosaic/scripts/cli.py:102]: I ran black, but maybe it didn't propagate?

Comment (kylebarron) [reply cogeo_mosaic/scripts/cli.py:102]: Actually it's because of the `,` after `url`

File: cogeo_mosaic/scripts/cli.py (hunk 1 above)
Comment (kylebarron) [reply cogeo_mosaic/scripts/cli.py]: Done

**Comment by vincentsarago** [comment]
Open question: should we setup a test incase of existing table or existing file on s3 ?

**Comment by kylebarron** [comment]
> Edit: DynamoDB will error if the table already exist

Will it? I don't think so. I think the default now is that it'll overwrite.

**Comment by kylebarron** [comment]
It does error, but it's in a `try/except`:
https://github.com/developmentseed/cogeo-mosaic/blob/aa06bd11fabed99306e5b6205694267874af1ff0/cogeo_mosaic/backends/dynamodb.py#L125-L139

**Comment by kylebarron** [comment]
But regardless, those test cases are unrelated to the CLI specifically

**Comment by vincentsarago** [comment]
> But regardless, those test cases are unrelated to the CLI specifically

good point, I'll open an issue and see if we want to add `--force` option if the table or the file on s3 already exits

**Comment by kylebarron** [comment]
But note that with dynamodb, the url must be specified with the region here:
```
dynamodb://us-west-2/table_name
```
the shorthand
```
dynamodb:///table_name
```
only works when the region is in an environment variable

**Comment by vincentsarago** [comment]
> only works when the region is in an environment variable

I think it's fine, I don't really want to over complicate the CLI, DynamoDB is a more advanced backend. Maybe we can update https://github.com/developmentseed/cogeo-mosaic#mosaic-storage-backends to make sure people know about the default region

**Comment by vincentsarago** [comment]
thanks @kylebarron this is great. I'll merge it in master but not sure if I will cut a release right away!

**Pull request merged**